### ISSUE TITLE
Missing --once param in start.ps1

### ISSUE THREAD

**Issue by clma91** [MicrosoftDocs/azure-devops-docs#13172]
You describe how to [Create and build the Dockerfile](https://learn.microsoft.com/en-us/azure/devops/pipelines/agents/docker?view=azure-devops#start-the-image) for the Windows images. In this section you offer the start script `start.ps1` to include as an entrypoint for the docker image. Later in the section [Start the image](https://learn.microsoft.com/en-us/azure/devops/pipelines/agents/docker?view=azure-devops#start-the-image) you state:

> If you want a fresh agent container for every pipeline run, pass the [--once flag](https://learn.microsoft.com/en-us/azure/devops/pipelines/agents/v2-windows?view=azure-devops#run-once) to the run command.

But this will actually not work with your provided start script because there is neither a param to passed to the start.ps1 nor to the run.cmd (start.ps1 line 59).

Please provide an accurate start script for agents running on Windows containers, like you do for Linux containers. Thanks!

---
#### Document Details

⚠ *Do not edit this section. It is required for learn.microsoft.com ➟ GitHub issue linking.*

* ID: b31d4f6c-d16d-2ba1-1294-6aa94e328046
* Version Independent ID: 88aec599-4410-697d-1002-1c41b1c17bb6
* Content: [Run a self-hosted agent in Docker - Azure Pipelines](https://learn.microsoft.com/en-us/azure/devops/pipelines/agents/docker?view=azure-devops)
* Content Source: [docs/pipelines/agents/docker.md](https://github.com/MicrosoftDocs/azure-devops-docs/blob/main/docs/pipelines/agents/docker.md)
* Service: **azure-devops-pipelines**
* Sub-service: **azure-devops-pipelines-agents**
* GitHub Login: @steved0x
* Microsoft Alias: **sdanie**

**Comment by WilliamAntonRohm** [comment]
@clma91 -- Claudio, thank you for your thorough feedback.  Please consider these resources:
- [Azure DevOps Services](https://azure.microsoft.com/support/devops/) 
- [Azure DevOps on Stack Overflow](https://stackoverflow.com/questions/tagged/azure-devops)

@steved0x -- Steve, please look into this issue.

**Comment by clma91** [comment]
@WilliamAntonRohm thanks for the quick response. I just noticed something in addition that might be linked to this issue. 

In contrast to the Linux agents, Windows agents won't de-register from Azure DevOps if the container is stopped/removed by the container runtime (e.g. Docker).
So, if I start a Windows agent in a container, it will register at Azure DevOps. If I stop the container after a while, it will not de-register at Azure DevOps but will obviously become offline as it looses connection. But it stays forever in Azure DevOps until it will be started again under the exact same agent name.

On the other hand, a Linux agent started in a container de-registers itself right away after it is stopped.

As a workaround I adapted for the start.ps1 with the `.\run.cmd --once` call. I noticed that after a pipeline ran with the `--once` flag set, it worked perfectly fine and de-registered the agent correctly:
```
2023-03-09 13:23:37Z: Listening for Jobs
2023-03-09 13:23:39Z: Running job: build_application
2023-03-09 13:25:21Z: Job build_application completed with result: Succeeded
Cleanup. Removing Azure Pipelines agent...
Removing agent from the server
Connecting to server ...
Succeeded: Removing agent from the server
Removing .credentials
Succeeded: Removing .credentials
Removing .agent
Succeeded: Removing .agent
```
